diff --git a/dist/modal.js b/dist/modal.js
index 80f4e751b165e9730cc74eae7d3cba9b6aef0f6f..313fd1bea862284b581e165458b62512d64c24ff 100644
--- a/dist/modal.js
+++ b/dist/modal.js
@@ -18,6 +18,7 @@ const defaultProps = {
     backdropOpacity: 0.7,
     backdropTransitionInTiming: 300,
     backdropTransitionOutTiming: 300,
+    backhandlerListener: null,
     customBackdrop: null,
     useNativeDriver: false,
     deviceHeight: null,
@@ -453,10 +454,13 @@ export class ReactNativeModal extends React.Component {
         if (this.state.isVisible) {
             this.open();
         }
-        BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        this.backhandlerListener = BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
     }
     componentWillUnmount() {
-        BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
+        if(this.backhandlerListener) {
+            this.backhandlerListener.remove();
+        }
+
         if (this.didUpdateDimensionsEmitter) {
             this.didUpdateDimensionsEmitter.remove();
         }
